common/tmx_filetype.go

```diff
@@ -0,0 +1,62 @@
+package common
+
+import (
+	"bytes"
+	"engo.io/engo"
+	"fmt"
+	"io"
+)
+
+// TMXResource contains a level created from a Tile Map XML
+type TMXResource struct {
+	Level *Level
+	url   string
+}
+
+// URL retrieves the url to the .tmx file
+func (r TMXResource) URL() string {
+	return r.url
+}
+
+// tmxLoader is responsible for managing '.tmx' files within 'engo.Files'.
+// You can generate a TMX file with the Tiled map editor.
+type tmxLoader struct {
+	levels map[string]TMXResource
+}
+
+// Load will load the tmx file and any other image resources that are needed
+func (t *tmxLoader) Load(url string, data io.Reader) error {
+	buf := new(bytes.Buffer)
+	_, err := buf.ReadFrom(data)
+	if err != nil {
+		return err
+	}
```

Comment:
While I'm perfectly fine with this, since you're only using the `[]byte` representation, this might be of use:

```go
b, err := ioutil.ReadAll(data)
if err != nil {
    return err
}

// and now `b` is of type `[]byte`
```